### PR TITLE
Always ignore freqs_cis

### PR DESCRIPTION
**Pull request by fegin**
We should always ignore freq_cis and other parameters in excluded_parameters_for_model_only to avoid confusion.

TODO: Is this going to break PP with seed checkpoint?

### CODE REVIEW

**Comment by tianyu-l** [comment]
> TODO: Is this going to break PP with seed checkpoint?

What's the reason for breaking PP?
Currently init_weights (including the buffer) is called, regardless of whether loading from a checkpoint (which doesn't sound optimal). So shouldn't cause issues?

File: torchtitan/components/checkpoint.py [hunk 1]
```diff
@@ -52,9 +52,16 @@ class AsyncMode(str, enum.Enum):
...
         self.model = [model] if isinstance(model, nn.Module) else model
-        self.cache_state_dict = {
+        self.cache_state_dict = self._get_state_dict()
+
+    def _get_state_dict(self) -> dict[str, Any]:
+        state_dict = {
             k: v for sd in map(get_model_state_dict, self.model) for k, v in sd.items()
         }
+        # Exclude parameters that should not be saved
+        for excluded_key in excluded_parameters_for_model_only:
```
Comment (tianyu-l) [inline torchtitan/components/checkpoint.py:68]: let's define `excluded_parameters_for_model_only` before it's used in this class for better readability

shall we also remove https://github.com/pytorch/torchtitan/blob/main/scripts/convert_llama_to_dcp.py#L127

**Comment by fegin** [comment]
> What's the reason for breaking PP?
> Currently init_weights (including the buffer) is called, regardless of whether loading from a checkpoint (which doesn't sound optimal). So shouldn't cause issues?

Not talking about the issue that there will be an exception. I'm wondering do we still need freqs_cis in the seed checkpoint to ensure that freqs_cis being consistent across PP stages?

**Review by tianyu-l — APPROVED** [review]
> I'm wondering do we still need freqs_cis in the seed checkpoint to ensure that freqs_cis being consistent across PP stages?

I don't think we need it in the seed checkpoint.
Every PP rank would init the same `freqs_cis` before loading a checkpoint
https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama3/model/model.py#L372

**Comment by tianyu-l** [comment]
The test generate script failed because it's not using `CheckpointManager`, but using `dcp.load` directly. Shall we change it and also use `CheckpointManager`?

**Comment by fegin** [comment]
Okay, let's merge the PR since seed checkpoint doesn't seem to be a concern.
